File: string_datatype.py
# string_datatype.py

# Defining strings with type annotations (in Python, type hints can be added for clarity)
greeting = "Hello"  # Type: str
name = "Alice"      # Type: str

# Concatenating strings
message = greeting + " " + name
print(message)  # Output: "Hello Alice"

# String slicing: Extracting a part of the string
print(name[1:4])  # Output: "lic"

# String methods:
print(message.upper())  # Output: "HELLO ALICE"
print(message.lower())  # Output: "hello alice"
print(message.replace("Alice", "Bob"))  # Output: "Hello Bob"

# Checking if a substring exists within a string
if "Alice" in message:
    print("Alice is in the message.")

# String length: Calculating the length of the string
print("Length of the message:", len(message))  # Output: Length of the message: 11

File: string_datatype.ts
// string_datatype.ts
export {};

/**
 * This program demonstrates basic string operations in TypeScript.
 * It shows string concatenation, slicing, methods, substring checking, and string length calculation.
 */

// Defining strings with type annotations
const greeting: string = "Hello";  // Type annotation for string
const name: string = "Alice";      // Type annotation for string

// Concatenating strings
const message: string = greeting + " " + name;
console.log(message);  // Output: "Hello Alice"

// String slicing (TypeScript doesn't support slicing directly, so we use substring)
console.log(name.substring(1, 4));  // Output: "lic"

// String methods
console.log(message.toUpperCase());  // Output: "HELLO ALICE"
console.log(message.toLowerCase());  // Output: "hello alice"
console.log(message.replace("Alice", "Bob"));  // Output: "Hello Bob"

// Checking for substring
if (message.includes("Alice")) {
    console.log("Alice is in the message.");
}

// String length
console.log("Length of the message:", message.length);  // Output: Length of the message: 11